tests/flow/test_procedures.py

```diff
@@ -319,3 +319,11 @@ def test_procedure_lookup(self):
             # This should not cause an error
             self.env.assertFalse(1)
             pass
+
+    def test_non_existing_procedure(self):
+        try:
+            redis_graph.call_procedure("dB.non_existing_procedure")
+            self.env.assertFalse(1)
+        except redis.exceptions.ResponseError as e:
+            # Expecting a runtime error.
+            self.env.assertIn("is not registered", e.message)
```

Comment:
@DvirDukhan 
is this the expected error message?
See `QueryCtx_SetError("Unable to find a procedure with name %s", proc_name);`
or is it that we also emit another error which overwrite the current one?

Reply:
Well, we actually emit it on the AST validation, before the AST read-only check.
The problem was that the current lookup was case sensitive, meaning, I was looking for a procedure
`propertyKeys`
The AST validation found it, as `Proc_Get` is case insensitive, so the AST validation passed with success.
However, checking if the AST is read-only did not find `propertyKeys`, since was case sensitive. 
Anyway, I suggest we keep the error here, as every undefined behavior should trigger an error IMO.